WordPress/Classes/Services/BlogJetpackSettingsService.swift

```diff
@@ -134,20 +134,23 @@ struct BlogJetpackSettingsService {
                 return
         }
 
+        let changes = blogSettings.changedValues()
         let remote = BlogJetpackSettingsServiceRemote(wordPressComRestApi: remoteAPI)
-        remote.updateJetpackMonitorSettingsForSite(blogDotComId,
-                                                   settings: jetpackMonitorsSettingsRemote(blogSettings),
-                                                   success: {
-                                                       do {
-                                                           try self.context.save()
-                                                           success()
-                                                       } catch let error as NSError {
-                                                           failure(error)
-                                                       }
-                                                   },
-                                                   failure: { (error) in
-                                                       failure(error)
-                                                   })
+        remote.updateJetpackMonitorSettingsForSite(
+            blogDotComId,
+            settings: jetpackMonitorsSettingsRemote(blogSettings),
+            success: {
+                self.coreDataStack.performAndSave({ context in
+                    guard let blogSettings = Blog.lookup(withObjectID: blog.objectID, in: context)?.settings else {
+                        return
+                    }
+                    for (key, value) in changes {
+                        blogSettings.setValue(value, forKey: key)
+                    }
+                }, completion: success)
+            },
+            failure: failure
```

Comment:
We seem to do this block a few times, I wonder about extracting it into a method?

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/19920/commits/ca92e62e5e88039f438ee9bf34679faf52dc4f31